library/core/src/alloc/layout.rs

```diff
@@ -85,6 +91,17 @@ impl Layout {
         unsafe { Ok(Layout::from_size_align_unchecked(size, align)) }
     }
 
+    /// Internal helper constructor to skip revalidating alignment validity.
+    #[inline]
+    const fn from_size_valid_align(size: usize, align: ValidAlign) -> Result<Self, LayoutError> {
+        // See above for the correctness of this check.
+        if size > isize::MAX as usize - (align.as_nonzero().get() - 1) {
```

Comment:
I really wanted to offer a `minus_one()` method on `NonZeroU##` to use for things like this, but annoyingly LLVM normalizes `sub nuw %align, 1` to `add %align, -1`, dropping the extra flag, so it turns out to be useless :(